frontend/src/pages/PipelineDetails.test.tsx

```diff
@@ -550,6 +550,69 @@ describe('PipelineDetails', () => {
     );
   });
 
+  it(
+    'uses an empty string and does not show error ' +
+      'when pipeline_spec in the response of getPipelineVersion() is undefined',
```

Comment:
Probably we should call this `shows error when pipeline_spec in the response of getPipelineVersion() is undefined`